### PR TITLE
test: Add tests for CPS source connector, PSL sink and source connectors

### LINKED ISSUE #98: Initialization error in PubsubLiteSinkConnector
Bug was introduced in https://github.com/googleapis/java-pubsub-group-kafka-connector/pull/83. In PublisherFactoryImpl.java, [`getPartitionPublisherFactory` does not set value for required property `streamFactory`](https://github.com/googleapis/java-pubsub-group-kafka-connector/blob/243df288405327e2468fd8ad6220daa173599149/src/main/java/com/google/pubsublite/kafka/sink/PublisherFactoryImpl.java#L41-L47) causing Autovalue validation to fail on reflection and an uncaught exception to be raised. The ultimate result is that the `PubsubLiteSinkTask` is never properly initialized.

#### Steps to reproduce
1. Try to bring up a PubsubLiteSinkConnector.java task

#### Stack trace
```
1768-[2022-09-07 17:46:32,076] ERROR [PubSubLiteSinkConnector|task-0] WorkerSinkTask{id=PubSubLiteSinkConnector-0} Task threw an uncaught and unrecoverable exception. Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask:195)
1769:java.lang.IllegalStateException: Missing required properties: streamFactory
1770-   at com.google.cloud.pubsublite.internal.wire.AutoValue_SinglePartitionPublisherBuilder$Builder.autoBuild(AutoValue_SinglePartitionPublisherBuilder.java:173)
1771-   at com.google.cloud.pubsublite.internal.wire.SinglePartitionPublisherBuilder$Builder.build(SinglePartitionPublisherBuilder.java:67)
1772-   at com.google.pubsublite.kafka.sink.PublisherFactoryImpl$1.newPublisher(PublisherFactoryImpl.java:47)
1773-   at com.google.cloud.pubsublite.internal.wire.RoutingPublisherBuilder$Builder.build(RoutingPublisherBuilder.java:69)
1774-   at com.google.pubsublite.kafka.sink.PublisherFactoryImpl.newPublisher(PublisherFactoryImpl.java:69)
1775-   at com.google.pubsublite.kafka.sink.PubSubLiteSinkTask.start(PubSubLiteSinkTask.java:64)
1776-   at org.apache.kafka.connect.runtime.WorkerSinkTask.initializeAndStart(WorkerSinkTask.java:312)
1777-   at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:186)
1778-   at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:243)
1779-   at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
1780-   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
1781-   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
1782-   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
1783-   at java.base/java.lang.Thread.run(Thread.java:829)
1784-[2022-09-07 17:46:32,079] WARN [PubSubLiteSinkConnector|task-0] Could not stop task (org.apache.kafka.connect.runtime.WorkerSinkTask:173)
1785-java.lang.IllegalStateException: Called stop when publisher doesn't exist.
1786-   at com.google.pubsublite.kafka.sink.PubSubLiteSinkTask.stop(PubSubLiteSinkTask.java:138)
1787-   at org.apache.kafka.connect.runtime.WorkerSinkTask.close(WorkerSinkTask.java:171)
1788-   at org.apache.kafka.connect.runtime.WorkerTask.doClose(WorkerTask.java:167)
1789-   at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:199)
1790-   at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:243)
1791-   at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
1792-   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
1793-   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
1794-   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
1795-   at java.base/java.lang.Thread.run(Thread.java:829)
1796-[2022-09-07 17:46:32,079] ERROR [PubSubLiteSinkConnector|task-1] WorkerSinkTask{id=PubSubLiteSinkConnector-1} Task threw an uncaught and unrecoverable exception. Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask:195)
1797:java.lang.IllegalStateException: Missing required properties: streamFactory
1798-   at com.google.cloud.pubsublite.internal.wire.AutoValue_SinglePartitionPublisherBuilder$Builder.autoBuild(AutoValue_SinglePartitionPublisherBuilder.java:173)
1799-   at com.google.cloud.pubsublite.internal.wire.SinglePartitionPublisherBuilder$Builder.build(SinglePartitionPublisherBuilder.java:67)
1800-   at com.google.pubsublite.kafka.sink.PublisherFactoryImpl$1.newPublisher(PublisherFactoryImpl.java:47)
1801-   at com.google.cloud.pubsublite.internal.wire.RoutingPublisherBuilder$Builder.build(RoutingPublisherBuilder.java:69)
1802-   at com.google.pubsublite.kafka.sink.PublisherFactoryImpl.newPublisher(PublisherFactoryImpl.java:69)
1803-   at com.google.pubsublite.kafka.sink.PubSubLiteSinkTask.start(PubSubLiteSinkTask.java:64)
1804-   at org.apache.kafka.connect.runtime.WorkerSinkTask.initializeAndStart(WorkerSinkTask.java:312)
1805-   at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:186)
1806-   at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:243)
1807-   at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
1808-   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
1809-   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
1810-   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
1811-   at java.base/java.lang.Thread.run(Thread.java:829)
1812-[2022-09-07 17:46:32,080] WARN [PubSubLiteSinkConnector|task-1] Could not stop task (org.apache.kafka.connect.runtime.WorkerSinkTask:173)
1813-java.lang.IllegalStateException: Called stop when publisher doesn't exist.
1814-   at com.google.pubsublite.kafka.sink.PubSubLiteSinkTask.stop(PubSubLiteSinkTask.java:138)
1815-   at org.apache.kafka.connect.runtime.WorkerSinkTask.close(WorkerSinkTask.java:171)
1816-   at org.apache.kafka.connect.runtime.WorkerTask.doClose(WorkerTask.java:167)
1817-   at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:199)
1818-   at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:243)
1819-   at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
1820-   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
1821-   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
1822-   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
1823-   at java.base/java.lang.Thread.run(Thread.java:829)
```

### PR DESCRIPTION
**Pull request by samarthsingal**
* Adds integration tests for the CPS source connector and PSL source and sink connectors
* Fixes PublisherSettings initialization issue in PubsubLiteSinkTask that caused fatal exception on running

### CODE REVIEW

**Review by hannahrogers-google — APPROVED** [review]

**Review by anguillanneuf — CHANGES_REQUESTED** [review]
Thanks Samarth for adding these tests!

Missing license headers in the following files: 
- src/test/java/it/StandaloneIT.java
- src/test/java/it/Base.java

Group test/compile/provided dependencies in pom.xml. 
Place test dependencies beneath compile/provided deps in pom.xml. 

Can we use `google-cloud-pubsub-bom` and `google-cloud-pubsublite-bom` instead of pulling in individual deps that are part of those bom's?

File: src/test/java/it/StandaloneIT.java [hunk 1]
```diff
@@ -254,55 +431,245 @@ public void testCpsSinkConnector() throws Exception {
...
+    // Consume from Kafka connect.
+    Properties consumer_props = new Properties();
+    consumer_props.setProperty(
+        ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaInstanceIpAddress + ":" + KAFKA_PORT);
+    consumer_props.setProperty(
+        ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+        "org.apache.kafka.common.serialization.StringDeserializer");
+    consumer_props.setProperty(
+        ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+        "org.apache.kafka.common.serialization.StringDeserializer");
```
[the remark this answers is not in the document]
Comment (samarthsingal) [reply src/test/java/it/StandaloneIT.java:477]: We write a string message so we want to deserialize it as a string to read. Note that this is different from the key/value serializer used in the connector configs which are ByteArrayConverters so the on-disk encoding of message data in Kafka is bytes.

File: src/test/java/it/StandaloneIT.java [hunk 2]
```diff
@@ -254,55 +431,245 @@ public void testCpsSinkConnector() throws Exception {
...
+                    FlowControlSettings.builder()
+                        .setBytesOutstanding(10 * 1024 * 1024L)
+                        .setMessagesOutstanding(1000L)
+                        .build())
+                .build());
+    try {
+      subscriber.startAsync().awaitRunning();
+      subscriber.awaitTerminated(3, TimeUnit.MINUTES);
+    } catch (TimeoutException timeoutException) {
+      // Shut down the subscriber after 30s. Stop receiving messages.
```
[the remark this answers is not in the document]
Comment (samarthsingal) [reply src/test/java/it/StandaloneIT.java:573]: Changed to 3min.

File: src/test/resources/kafka_vm_startup_script.sh [hunk 3]
```diff
@@ -43,7 +49,13 @@ sed -i "s@#plugin.path=@plugin.path=$(pwd)\/$GCS_DIR@g" $KAFKA_DIR/config/connec
...
+$KAFKA_DIR/bin/kafka-topics.sh --create --topic 'psl-source-test-topic' --bootstrap-server localhost:9092
 ## Start connectors
-$KAFKA_DIR/bin/connect-standalone.sh $KAFKA_DIR/config/connect-standalone.properties $GCS_DIR/$CPS_SINK_CONNECTOR_PROPERTIES $GCS_DIR/$CPS_SOURCE_CONNECTOR_PROPERTIES &
+$KAFKA_DIR/bin/connect-standalone.sh $KAFKA_DIR/config/connect-standalone.properties \
+ $GCS_DIR/$CPS_SINK_CONNECTOR_PROPERTIES \
+ $GCS_DIR/$CPS_SOURCE_CONNECTOR_PROPERTIES \
+ $GCS_DIR/$PSL_SINK_CONNECTOR_PROPERTIES \
+ $GCS_DIR/$PSL_SOURCE_CONNECTOR_PROPERTIES &
 
 set +x
```
[the remark this answers is not in the document]
Comment (samarthsingal) [reply src/test/resources/kafka_vm_startup_script.sh]: Done.

File: src/test/resources/log4j.properties [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Root logger option
+log4j.rootLogger=INFO, stdout
+
+# Direct log messages to stdout
+log4j.appender.stdout=org.apache.log4j.ConsoleAppender
+log4j.appender.stdout.Target=System.out
+log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
+log4j.appender.stdout.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1} - %m%n
```
[the remark this answers is not in the document]
Comment (samarthsingal) [reply src/test/resources/log4j.properties]: Done.

File: src/test/resources/pubsub-lite-sink-connector-test.properties [hunk 5]
```diff
@@ -1,7 +1,9 @@
...
-pubsublite.location=europe-south7-q
-pubsublite.topic=my-topic
\ No newline at end of file
+tasks.max=2
+topics=psl-sink-test-topic
+pubsublite.project=<projectName>
+pubsublite.location=<pslZone>
+pubsublite.topic=psl-sink-topic-<runId>
+key.converter=org.apache.kafka.connect.converters.ByteArrayConverter
+value.converter=org.apache.kafka.connect.converters.ByteArrayConverter
```
[the remark this answers is not in the document]
Comment (samarthsingal) [reply src/test/resources/pubsub-lite-sink-connector-test.properties]: Done.

File: src/test/resources/pubsub-lite-source-connector-test.properties [hunk 6]
```diff
@@ -1,7 +1,9 @@
...
-pubsublite.subscription=my-subscription
-kafka.topic=my-kafka-topic
\ No newline at end of file
+tasks.max=2
+pubsublite.project=<projectName>
+pubsublite.location=<pslZone>
+pubsublite.subscription=psl-source-subscription-<runId>
+kafka.topic=psl-source-test-topic
+key.converter=org.apache.kafka.connect.converters.ByteArrayConverter
+value.converter=org.apache.kafka.connect.converters.ByteArrayConverter
```
[the remark this answers is not in the document]
Comment (samarthsingal) [reply src/test/resources/pubsub-lite-source-connector-test.properties]: Done.

File: src/test/java/it/StandaloneIT.java [hunk 7]
```diff
@@ -254,55 +431,245 @@ public void testCpsSinkConnector() throws Exception {
...
+          public void onPartitionsAssigned(Collection<TopicPartition> collection) {
+            assignmentReceived[0] = true;
+          }
+        });
+    LocalTime startTime = LocalTime.now();
+    boolean messageReceived = false;
+    try {
+      while (Duration.between(startTime, LocalTime.now())
+              .compareTo(Duration.of(2, ChronoUnit.MINUTES))
+          < 0) {
```
[the remark this answers is not in the document]
Comment (samarthsingal) [reply src/test/java/it/StandaloneIT.java:500]: Done.

**Review by anguillanneuf — APPROVED** [review]

**Pull request merged**